R/stacked_barchart.R

```diff
@@ -0,0 +1,44 @@
+#' stacked_barchart
+#' stacked barchart of the top_n features in a relative abundance table
+#'
+#' @param abundance_matrix dataframe; samples are columns and features are rows (relative abundance)
+#' @param top_n integer; number of taxa to plot in the stacked barchart. Default is 10.
+#' @import ggplot2
+#' @import reshape2
+#' @import RColorBrewer
+#' @export
+#' @return grob
+#' @examples
+#' stacked_barchart()
+
+stacked_barchart <- function(abundance_matrix, top_n = 10){
+  
+  # The top_n is defined by the average relative abundance
+  # across samples in the matrix
+  ave_abund <- rowMeans(log10(abundance_matrix + 1e-6))
+  relabund <- abundance_matrix[order(ave_abund, decreasing=TRUE),]
+  
+  # take the top n
+  toppers <- rownames(relabund[1:top_n,])
```

Comment:
no need to change. Just leaving a note for myself to apply top_taxa() here once I add top_taxa() to the package